crates/consensus-logic/src/csm/client_transition.rs

```diff
@@ -353,22 +353,22 @@ mod tests {
     use super::*;
     use crate::genesis;
 
-    struct TestEvent {
+    struct TestEvent<'a> {
         event: SyncEvent,
-        expected_writes: Vec<ClientStateWrite>,
-        expected_actions: Vec<SyncAction>,
+        expected_writes: &'a [ClientStateWrite],
+        expected_actions: &'a [SyncAction],
```

Comment:
I think that these make more sense to be inside `TestCase` struct.
Since these expected values will eventually be used to assert things, I think they be together with `state_assertions`.

What do you think?

Reply:
Ah, I see it now after a discussion, these expected values are meant to run for each event and state assertions are meant to run after processing all of the events.

Reply:
Yeah, that's right.